### DISCUSSION TITLE
Copilot Chat stopped working on Visual Studio 2022

### DISCUSSION THREAD

**Discussion by zdong** [community/community#68864] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I installed copilot chat for two days. First day it works fine, and second day it stopped working. 
![image](https://github.com/community/community/assets/3258604/4f4185cd-02f5-4860-99de-c83c0ec14def)
Copilot still works, just not Copilot chat on VS2022.

Diabled and enabled, uninstalled and reinstalled. It still show:
"There was a problem completing your request. please try again"
System info:
Windows 10 Pro: Build 19044
VS2022:  VisualStudio.17.Release/17.7.4+34031.279
GitHub Copilot: 1.110.0.0 (v1.110.0.0@9f24d0f53)
GitHub Copilot Agent: 1.110.389 (v1.110.0)
Copilot Chat: 0.1.1386.44816

On the same machine, copilot chat works fine on VSCode:
VScode: Version: 1.82.2
Copilot: v1.121.453
Copilot chat: v0.7.1

**Comment by WiseyD** [comment]
Having the same issue, Visual Studio Professional version 17.7.4, also tried disabling, re-enabling, logging out and back in, re-authorizing  un-installed, re-installed.

**Comment by EmbyrCelyca** [comment]
I'm having the same issue .  Looks like all the same details as above as well. I am currently on a trial for copilot, not sure if that could be it.

**Comment by mavaddat** [comment]
Same issue. Copilot Chat works in Visual Studio Code, but not in Visual Studio.

From the `GitHub Copilot chat` output from Visual Studio:

```
Logging to: C:\Users\B0649033\AppData\Local\Temp\VSGitHubCopilotLogs\20230930_230128_VSGitHubCopilot.chat.log
[Conversations Information] Copilot chat version 0.1.1386-beta+af10cc9358 (0.1.1386.44816). VS: VisualStudio.17.Preview/17.8.0-pre.2.0+34112.27. Session: ████████-████-████-████-████████████. 
[ConversationsServiceManager Information] Attempting to get out-of-process service proxy.
[ConversationsServiceManager Information] Out-of-process service proxy obtained.
[ConversationsServiceManager Information] Out-of-process service finished initializing.
[Conversations Information] [CopilotClient] Need to get an auth token.
[Conversations Information] [CopilotClient] Need to get an auth token.
[Conversations Information] [CopilotClient] Status Code: 200, Reason Phrase: OK
[Conversations Information] [CopilotClient] Response Content: HasToken: True, ChatEnabled: True, ExpiresAt: 1696116692, RefreshIn: 1500, SKU: monthly_subscriber
[Conversations Information] [CopilotClient] Status Code: 200, Reason Phrase: OK
[Conversations Information] [CopilotClient] Response Content: HasToken: True, ChatEnabled: True, ExpiresAt: 1696116692, RefreshIn: 1500, SKU: monthly_subscriber
[Conversations Information] [CopilotClient] Obtained a token which is valid for 00:24:59.9999841.
[Conversations Information] [CopilotClient] Obtained a token which is valid for 00:24:59.9999712.
[Conversations Information] Current syntax node is a method.
[CopilotSessionProvider Information] Begin sending message (ConversationId:████████-████-████-████-████████████, CorrelationId:b93ef302-f48d-4984-86e1-ff88202c9150, MessageId: 05849dc4-ff84-4f55-8b30-07371f032bbc)
[CopilotSessionProvider Information] Request content: "██████████████████████████████"
```

Version info:

```
Microsoft Visual Studio Community 2022
Version 17.8.0 Preview 2.0
VisualStudio.17.Preview/17.8.0-pre.2.0+34112.27
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

ASP.NET and Web Tools   17.8.226.21692
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.8.226.21692
Azure App Service Tools v3.0.0

C# Tools   4.8.0-2.23429.7+44555193fd1135b5d53a2099f76fec91e0d1ebde
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GhostDoc   22.3.23200.0
Generate XML Comments from your code, maintain clean and up-to-date documentation, produce help documentation in multiple formats, use intelligent source code Spell Checker in Visual Studio.

GitHub Copilot   1.110.0.0 (v1.110.0.0@9f24d0f53)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.110.389 (v1.110.0)

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.8.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Oracle Developer Tools for Visual Studio   21.7.0.0
Oracle Developer Tools for Visual Studio Copyright (c) 2005, 2022

PostSharp Tools for Visual Studio   2023.0.8.0
PostSharp is the #1 pattern-aware extension to C# and VB.

Razor (ASP.NET Core)   17.8.2.2345506+ade90399d42c1a7bf92191b1c067816c0ae1c311
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.8.64.0
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.20830.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.8.0-2.23429.7+44555193fd1135b5d53a2099f76fec91e0d1ebde
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.8.0-beta.23425.10+0d3549fa5b8b6387ade191d76768405cefed8229
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.
```

**Comment by Warsen** [comment]
I too am having this problem. It started around 24 hours ago while I was in the middle of asking it something and it has been broken since.

**Reply by Lvl6Technician** [reply]
![image](https://github.com/community/community/assets/109123426/6853cad2-2979-4e49-8d2f-61e8ad245ed8)

Same here, same downtime timeframe.

**Comment by Zercho75** [comment]
Mismo problema en visual studio, en VSC si funciona

**Comment by zenerbufen** [comment]
I'm getting this error as well. It works fine in vscode, but does not work in VS 2022

**Comment by ghhv** [comment]
Same here.. Stopped a few days ago. Have upgraded VS today. No fix. No errors. It is getting a token. Not working on VS2022 normal and Preview.

**Comment by Respawn84** [comment]
Same problem here. Not working on VS2022 17.7.0 Preview 4.0

**Comment by stefv** [comment]
Same problem: Microsoft Visual Studio Community 2022 (64-bit) - Current Version 17.7.4

**Comment by francisgariepy** [comment]
Same here, it was working yesterday and suddenly stopped this morning. I tried to uninstall and reinstall the VS 2022 extension but still giving the same error.

Visual Studio Community 2022 (64-bit) - Current (17.7.4)

**Comment by AleCie** [comment]
same problem, first time using copilot

**Comment by CheeseBard** [comment]
Same problem since 2 days ago. Payment completed and in use.
![image](https://github.com/community/community/assets/146434200/a856df64-e28f-45fd-82e7-12f3cea69ff7)

**Comment by DNin01** [comment]
Maybe it's https://github.blog/changelog/2023-09-25-breaking-change-to-github-copilot-service-endpoints?

**Reply by CheeseBard** [reply]
I added domain whitelist to my firewall but still have the same problem.

**Comment by RPossberg** [comment]
Having the same issue. Tried logging out and back in, requested new authorization token, but Copilot chat is broken. In both Preview 17.8.0 and Community 2022 17.7.4
![GithubCoPilot_Authorized](https://github.com/community/community/assets/122515790/cd1c92a0-5bcf-469a-8465-0779f8062436)

**Comment by longago2009** [comment]
Same problem since 2 days ago
Visual Studio Community 2022 (64-bit) - Current (17.7.4)

**Comment by Adiiiii88** [comment]
I have the same problem with VS 2022 Enterprise (17.6.2)

**Comment by dosimetrybadge** [comment]
People are reporting the same issue on the C# discord

**Comment by bmaluijb** [comment]
I have the same problem in both VS2022 regular, and VS2022 Preview, latest versions. Please solve this fast?

**Comment by Umutks** [comment]
Same problem 17.7.4 community edition copilot is not working.

**Comment by cyyeong** [comment]
same with VS2022 :(

**Comment by hubaksis** [comment]
Same issue here.
`
[CopilotSessionProvider Information] Begin sending message (ConversationId:xxxx, CorrelationId:dd9cd6f0-709d-4eb5-b450-c1c6cf4422dc, MessageId: ef8c8599-0ac3-413c-b6ef-b12af173fe9e)
[CopilotSessionProvider Information] Request content: "test"
`

**Comment by zlarumbe** [comment]
> ### Select Topic Area
> 
> Question
> 
> ### Body
> 
> I installed copilot chat for two days. First day it works fine, and second day it stopped working. 
> ![image](https://github.com/community/community/assets/3258604/4f4185cd-02f5-4860-99de-c83c0ec14def)
> Copilot still works, just not Copilot chat on VS2022.
> 
> Diabled and enabled, uninstalled and reinstalled. It still show:
> "There was a problem completing your request. please try again"
> System info:
> Windows 10 Pro: Build 19044
> VS2022:  VisualStudio.17.Release/17.7.4+34031.279
> GitHub Copilot: 1.110.0.0 (v1.110.0.0@9f24d0f53)
> GitHub Copilot Agent: 1.110.389 (v1.110.0)
> Copilot Chat: 0.1.1386.44816
> 
> On the same machine, copilot chat works fine on VSCode:
> VScode: Version: 1.82.2
> Copilot: v1.121.453
> Copilot chat: v0.7.1

**Reply by zlarumbe** [reply]
https://my-store-ea2d54.creator-spring.com/listing/online-sales-store-3721

**Reply by WiseyD** [reply]
I'm having issues with inline suggestions as well.

**Comment by joaorzz** [comment]
Same problem here in VS2022... I've been using vs code

**Reply by zlarumbe** [reply]
> Same problem here in VS2022... I've been using vs code

**Comment by LuisiitoDev** [comment]
Same problem VS 2022 17.7.4

**Comment by motus-uk-simon** [comment]
Same problem. Started a few days ago. Chat works in VS Code

**Comment by jmolinelli** [comment]
Same problem VS 2022 17.7.4. Hasn't worked for days

**Comment by vmartinezlcf** [comment]
I have the same issue, it started today. VS 2022 doesn't work, VS code does work.

**Comment by bmaluijb** [comment]
I think this is affecting many users, as I’m also seeing on X. I haven’t had access to Chat and inline for days now, and tried all of the above and more. Please prioritize this issue.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by bmaluijb** [comment]
Can you point us to the GitHub Copilot team please? I don’t know what their contact information is.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by MichelleXIE2023** [reply]
The same here. I am using visual studio 2022.

**Comment by jlcfg** [comment]
EDIT:  

I received this from support:

...as Copilot Chat is currently still in Beta, further support will need to come from the public Discussions forum...
I would suggest keeping an eye on this thread and/or adding your own experience.

**Reply by namelessmasses** [reply]
Isn't chat still in technical preview? Github support site says they don't support copilot features in technical preview and to visit... you guessed it... here

> We've recently announced several new technical previews for GitHub Copilot. Please note that GitHub support will not be able to answer any questions regarding features that are under technical preview. Please visit [the GitHub Copilot Community page](https://github.com/orgs/community/discussions/categories/copilot) to leave your questions or feedback and interact with the community.

It'll still let you create a ticket - which I did - I'm just not holding my breath on that one. 

At first the above quote also had a link to a form for VS2022 copilot chat support but now can't get it to show again. The form was marked as retired with a comment from, I think, Mark Wilson-Thomas (MS Principal Program Manager for Copilot Chat) said to use "Send Feedback" from within VS2022 now.

**Reply by jlcfg** [reply]
Looks like there are 3 related issues on the vs developer community site:

https://developercommunity.visualstudio.com/t/GitHub-Copilot-does-not-work---visual-st/10481829
https://developercommunity.visualstudio.com/t/Github-copilot-chat-dont-answer--There/10481868
https://developercommunity.visualstudio.com/t/copilot-chat-there-was-a-problem-complet/10481897

**Reply by namelessmasses** [reply]
Nice. Looks like 

https://developercommunity.visualstudio.com/t/Github-copilot-chat-dont-answer--There/10481868?q=%E2%80%9Ccopilot+chat%E2%80%9D+%E2%80%9Cthere+was+a+problem%E2%80%9D 

has the most promising reply, although it’s jus a bot replying. The others don’t have any.

**Comment by TacetDiscipulus** [comment, marked as answer]
IT'S ALIVE!!!!!!!!!!!!! 

I just saw the notice from Microsoft, it was posed about 20 minutes ago. I gave it a try and sure enough I got a response. Everything is right in the world again!

Somewhat alarming though is how much I felt I lost without it. It felt almost paralyzing. I could code just fine before copilot, I can still code at the same level so really I was no worse off than a few months ago but it sure felt like I was. 

I'd like to give a big thanks to GitHub and Microsoft for resolving this issue so quickly, especially over a weekend and for a product that is in preview with no service agreements. It's easy to hear people hate on Microsoft but as far as their dev tools are concerned they are committed to constant and steady improvement. Also the few times I have required technical support their service I got was top notch. 

T

**Reply by WiseyD** [reply]
Looks to be resolved for me as well.

**Reply by RPossberg** [reply]
Yes, it is working again. :)

**Reply by fx-vera** [reply]
It's broken again since yesterday =(
<img width="323" alt="image" src="https://github.com/community/community/assets/14359283/9e1b9aff-b649-49fb-80d8-3afa87f1a30a">

**Reply by g-fleming** [reply]
I don't understand how this is marked as resolved. There is no information on the fix. I get this error when attempting to use copilot chat in either Visual Studio 2022 17.7.5 or VS Preview 2022 17.8:
[CopilotSessionProvider Information] Begin sending message (ConversationId:1111111, CorrelationId:1111111, MessageId: 1111111111111)
[CopilotSessionProvider Information] Request content: "What does Console.WriteLine() do?"
[Conversations Error] [CopilotClient] Failed to talk to Copilot: Response status code does not indicate success: 400 (Bad Request).

**Reply by fx-vera** [reply]
It seems the service was down because I had the same problem 2 weeks ago .
It was restored for 2 weeks but again it is failing (**I have the paid version**)
I'm using VS2022 latest update. Github account OK. VS restarted, etc, etc.

LOG:
[Conversations Information] [CopilotClient] Duration: 184
[Conversations Verbose] [CopilotClient] Raw response from Copilot: CopilotClientException:::bad request: missing Editor-Version header for IDE auth

[InteractiveSessionProvider Information] Response Type: Markdown
[InteractiveSessionProvider Information] Response: Sorry, error fulfilling request. Details: ::bad request: missing Editor-Version header for IDE auth

[InteractiveSessionProvider Information] Snippet Edit Type: Unknown

**Comment by CheeseBard** [comment]
Confirmed it's working again. The time without a copilot was truly terrible :) I felt like a civilized person in a ruined world who suddenly lost their technological capabilities, and it made me think a lot about AI.

**Comment by Abkeyev99** [comment]
Thanks so good :)

**Comment by bmaluijb** [comment]
Fixed for me too! Thank you very much. Like others have said: I just realized the importance of this product to me. I actually delayed several complex problems until it worked again, just to make it easier for myself to solve these things. What a crazy world we live in, as a year ago, none of this existed.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by PeteBaker1** [comment]
I am having problems with github copilot chat not working as I am a new user and it asks me to sign up for the preview, but the preview has ended? so the link it suggests is broken now!

**Comment by sohailaboebs** [comment]
I have the same problem, I pay my subscription fees, I opened a ticket and they directed me to this community, the answer was :"Please note that GitHub Support will not be able to answer questions regarding features under technical preview, including Copilot Chat. " 

I have Microsoft Visual Studio Community 2022 (64-bit) - Current
Version 17.7.5

**Reply by sohailaboebs** [reply]
They updated Visual Studio today and it works fine now, I guess this was a bug with the previous version of Visual Studio, the new version is Version 17.7.6

**Reply by fx-vera** [reply]
Thank you, I'm going to update the VS.

**Comment by g-fleming** [comment]
It is working for me now. I suppose it was just down. Perhaps it is not 100% up while in Beta.

**Reply by fx-vera** [reply]
Keeps failing for me =(

**Reply by sohailaboebs** [reply]
They updated Visual Studio today and it works fine now, I guess this was a bug with the previous version of Visual Studio, the new version is Version 17.7.6

**Reply by sohailaboebs** [reply]
thank you

**Reply by pdinbkk** [reply]
On VS2022 17.7.7 - still not working

**Reply by puppetSpace** [reply]
not working for me in VS2022 Version 17.8.0 Preview 5.0
In VsCode it is working

**Reply by ewaszoeke** [reply]
not working in Visual Studio Version 17.7.6

**Reply by g-fleming** [reply]
I think it just is unreliable right now. Sometimes it works for me sometimes it doesn't (day to day). I suspect my issue of server response error is not directly related to the VS2022 version.

**Comment by jamtrac-upenn** [comment]
This is not working at all for me. [Conversations Error] [CopilotClient] Failed to talk to Copilot: Response status code does not indicate success: 400 (Bad Request) on every prompt

**Reply by fx-vera** [reply]
I'm in the same situation. I'm going to unsubscribe.

**Reply by hubaksis** [reply]
Did you try reinstalling and do the auth again for Copilot? 
Does it say anything in the GitHub Copilot Output?

**Reply by fx-vera** [reply]
I will try reinstalling again, but yes, i have tried reinstalling and resetting the credentials. Also upgrading VS version.
The log is bad request in IDE.

Anyway, it is not the first time that I have problems, one month ago happened the same.

**Reply by jamtrac-upenn** [reply]
There is nothing in Github Copilot output. There is the bad request message I described above in Github Copilot output. Like fx-vera I have uninstalled, reinstalled, logged out and in, and upgraded VS 2022. No luck. Very frustrating, as this started right when I was scheduled to demo this to my team.

**Comment by ewaldatsensentia** [comment]
I have Copilot installed and working correctly.  Copilot Chat does not authenticate no matter what I do - uninstalling, logging in and out of my GitHub account...  I am using VS Community version 17.8.2.  Is this a new issue, or more of the above?
The GitHub Copilot chat output is:
[Conversations Information] [CopilotClient] Need to get an auth token.
[Conversations Information] [CopilotClient] Need to get an auth token.
[Conversations Information] [CopilotClient] Status Code: 401, Reason Phrase: Unauthorized
[Conversations Information] [CopilotClient] Response Content: HasToken: False, ChatEnabled: , ExpiresAt: 0, RefreshIn: 0, SKU: 
[Conversations Error] [CopilotClient] User's credentials need to be refreshed. Please sign out and sign back in to GitHub. https://learn.microsoft.com/en-us/visualstudio/ide/work-with-github-accounts?view=vs-2022
[Conversations Information] [CopilotClient] Status Code: 401, Reason Phrase: Unauthorized

**Comment by mnazers734** [comment]
Github Co-pilot chat is very unreliable in VS2022. Seems to work most the time in VS Code. I though the point of CoPilot chat was that I didn't have to leave my workspace to ask questions.

**Comment by christinemaulsby** [comment]
Tuesday Copilot Chat worked great in VS 2022 17.8. Today I'm receiving ConversationPackage could not be loaded error. This is greatly affecting my work as I depend heavily on Copilot Chat.

Note: I've uninstalled/reinstalled Copilot and Copilot Chat; I repaired VS; I upgraded to 17.9; i cleared cache; I reset general settings. None have resolved the issue.

**Reply by ayim** [reply]
Hey! If you're having issues, please use the "Report a Problem" feature in Visual Studio.
This provides a bunch of diagnostics to help us investigate

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
aayim@microsoft.com

**Reply by ayim** [reply]
Moved the conversation to a fresh thread here: https://github.com/orgs/community/discussions/114652

**Comment by ayim** [comment]
Locking this conversation since it seems to have a mix of stale issues and attract a variety of unrelated issues.
If you're having issues with GitHub Copilot in Visual Studio, I recommend filing a new discussion with a specific description about what's not working (i.e. ConversationPackage errors, error message upon install/load, etc)
